uni-stark/src/prover.rs

```diff
@@ -9,63 +8,110 @@ use p3_field::{
     cyclic_subgroup_coset_known_order, AbstractExtensionField, AbstractField, Field, PackedField,
     TwoAdicField,
 };
-use p3_matrix::dense::RowMajorMatrix;
+
+use p3_matrix::dense::{RowMajorMatrix};
 use p3_matrix::{Matrix, MatrixGet, MatrixRows};
+
 use p3_maybe_rayon::{IndexedParallelIterator, MaybeIntoParIter, ParallelIterator};
-use p3_util::log2_strict_usize;
+use p3_util::{log2_ceil_usize, log2_strict_usize};
 use tracing::{info_span, instrument};
 
-use crate::symbolic_builder::{get_log_quotient_degree, SymbolicAirBuilder};
+use crate::symbolic_builder::{SymbolicAirBuilder};
 use crate::{
-    decompose_and_flatten, Commitments, OpenedValues, Proof, ProverConstraintFolder, StarkConfig,
-    ZerofierOnCoset,
+    config, decompose_and_flatten, get_max_constraint_degree, Commitments, OpenedValues, Proof,
+    ProverConstraintFolder, StarkConfig, ZerofierOnCoset,
 };
+type Commitment<SC> = (
+    <<SC as config::StarkConfig>::Pcs as Pcs<
+        <SC as config::StarkConfig>::Val,
+        RowMajorMatrix<<SC as config::StarkConfig>::Val>,
+    >>::Commitment,
+    <<SC as config::StarkConfig>::Pcs as Pcs<
+        <SC as config::StarkConfig>::Val,
+        RowMajorMatrix<<SC as config::StarkConfig>::Val>,
+    >>::ProverData,
+);
+type ProverData<SC> = <<SC as config::StarkConfig>::Pcs as Pcs<
+    <SC as config::StarkConfig>::Val,
+    RowMajorMatrix<<SC as config::StarkConfig>::Val>,
+>>::ProverData;
 
-#[instrument(skip_all)]
-pub fn prove<
-    SC,
-    #[cfg(debug_assertions)] A: for<'a> Air<crate::check_constraints::DebugConstraintBuilder<'a, SC::Val>>,
-    #[cfg(not(debug_assertions))] A,
->(
+
+pub fn open<SC>(
     config: &SC,
-    air: &A,
```

Comment:
Is there a reason there are not type synonyms for these rather clunky type signatures in this function signature, namely for `trace_data`, `quotient_data`, and the return value?

Reply:
I have added type synonyms for ProverData and Commitment so function signatures should be clearer now